app/models/classified_listing.rb

```diff
@@ -62,8 +62,8 @@ class ClassifiedListing < ApplicationRecord
 
   scope :published, -> { where(published: true) }
 
-  def self.cost_by_category(category = "education")
-    categories_available[category][:cost]
+  def self.cost_by_category(category)
+    categories_available.dig(category, :cost) || 0
```

Comment:
Looks like `categories_available` will always come from [a frozen hash object](https://github.com/thepracticaldev/dev.to/blob/04825d361b13a39167568b7ff06c66e32a1fc310/app/models/classified_listing.rb#L8-L20); would we ever want to instead log or notify ourselves if someone is trying to retrieve a category without a `cost` on it (as opposed to falling back to `0` as the default here?)

Reply:
Although, now that I'm thinking about it, maybe that's weird to do for an API... 🤔

Reply:
An option could be to exit the method with a HTTP 422 as we do when params are invalid later on, let me look into it!